src/plugins/dashboard/public/application/top_nav/show_share_modal.tsx

```diff
@@ -151,19 +154,30 @@ export function ShowShareModal({
     ...unsavedStateForLocator,
   };
 
+  let _g = getStateFromKbnUrl<QueryState>('_g', window.location.href);
+  if (_g?.filters && _g.filters.length === 0) {
+    _g = _.omit(_g, 'filters');
+  }
+  const baseUrl = setStateToKbnUrl('_g', _g);
+
+  const shareableUrl = setStateToKbnUrl(
+    '_a',
+    stateToRawDashboardState({ state: unsavedDashboardState ?? {} }),
+    { useHash: false, storeInHashQuery: true },
+    unhashUrl(baseUrl)
+  );
+
+  const snapshotWarning = i18n.translate('dashboard.share.longUrlWarning', {
+    defaultMessage:
+      'This dashboard has a lot of unsaved changes, so it is recommended that you save before continuing.',
+  });
```

Comment:
**TAG DOCS ONCE PR OPEN FOR REVIEW**: Would really appreciate some help on the wording here :bow:

Reply:
@KOTungseth My only concern with that is that we only show the warning when **panel** changes are included in the unsaved state - so if, say, filters changed, controls changed, etc. the warning will not show up even if the dashboard is in an `unsaved changes` state. 

Currently, snapshot sharing is only really useful **if** there are unsaved changes in the first place - if there are no unsaved changes, then snapshot sharing and saved object sharing are virtually identical with this change. So, I'm not sure if pushing them to save the dashboard every time they generate a snapshot URL is the right move?

Would love your thoughts, though! 👍 Also cc @ThomThomson just to make sure my explanation is accurate 😆

Reply:
I'm fine if the error is more specific to panel changes, if @KOTungseth is good with that wording! I think since it only shows up for this specific case, then we might as well make the wording specific as well 👍 

As far as the subtle warning - very good point! This is what @andreadelrio suggested because the share menu already has a whole lot going on, but if we want it to be more obvious that this should be avoided, we could definitely make it stand out a bit more... 

Few ideas:
- We could set the `Snapshot` text to be a dark yellow along with the warning icon?
- We could go back to a more obvious error, like so? But the design is definitely not great so we would need @andreadelrio's opinion on it :) 

    ![image](https://user-images.githubusercontent.com/8698078/194104327-9760e9de-8ee3-45d8-b46b-bdd92f85e0a4.png)

- We could change the "Copy" button to yellow/warning state, and the popover could show up when highlighting there instead?

Reply:
New wording 👀  

![image](https://user-images.githubusercontent.com/8698078/194348781-3f93306d-bf8e-4d2e-bd06-debb7174a06d.png)

Looking forward to @andreadelrio's feedback on the possible placements of the warning 🙌

Reply:
@Heenawter I like the warning button!

Reply:
Seems like that way you can share whatever warning is relevant (for other link types as well)

Reply:
@andreadelrio Amazing, I'll go with the button! Thanks for your input! 🎉